### PR TITLE
Feat/new identities

### PR DESCRIPTION
**Pull request by JMounier**

### CODE REVIEW

**Comment by jeremad** [comment]
Please add the new `upgrade_identity` in the README

**Review by jeremad — APPROVED** [review]

File: tankersdk_identity/crypto/__init__.py [hunk 1]
```diff
@@ -27,6 +28,17 @@ def generichash(buffer: bytes, size: int) -> bytes:
     return generichash_blake2b_salt_personal(buffer, digest_size=size)  # type: ignore
 
 
+def hashed_provisional_email(email: str) -> str:
```
Comment (theodelrieu) [inline tankersdk_identity/crypto/__init__.py:31]: nitpick: I think I already commented on that before, but I'd have preferred `hash_provisional_email`.

We do perform an action here. Not that important though.

File: tankersdk_identity/test/test_identity.py [hunk 2]
```diff
@@ -77,6 +77,32 @@ def test_public_identity_matches_provisional_identity(test_app: Dict[str, str])
     assert public_identity["public_encryption_key"] == identity["public_encryption_key"]
 
 
+def test_public_identity_matches_sms_provisional_identity(
+    test_app: Dict[str, str]
+) -> None:
+    encoded_identity = tankersdk_identity.create_provisional_identity(
+        test_app["id"], "phone_number", "+611223344"
```
Comment (theodelrieu) [inline tankersdk_identity/test/test_identity.py:84]: Nitpick: I guess you meant "+33611223344"? We also have to patch those to use reserved numbers, but later.

File: tankersdk_identity/crypto/__init__.py (hunk 1 above)
Comment (tux3) [reply tankersdk_identity/crypto/__init__.py:31]: Yeah I think it's like the `sorted(x)` function in python. They have their own style sometimes...

File: tankersdk_identity/test/test_identity.py (hunk 2 above)
Comment (tux3) [reply tankersdk_identity/test/test_identity.py:84]: True. Will fix in the same PR that changes to reserved numbers.

**Review by theodelrieu — APPROVED** [review]

File: tankersdk_identity/test/test_identity.py [hunk 3]
```diff
@@ -105,3 +131,13 @@ def test_mistmatch_app_id_and_secret(test_app: Dict[str, str]) -> None:
...
+
+
+def test_invalid_provisional_target(test_app: Dict[str, str]) -> None:
+    with pytest.raises(ValueError):
+        tankersdk_identity.create_provisional_identity(
+            test_app["id"], "extremely invalid", "value"
+        )
+        tankersdk_identity.create_provisional_identity(
+            test_app["id"], "extremely invalid", "value"
+        )
```
Comment (blastrock) [inline tankersdk_identity/test/test_identity.py:143]: why twice? XD

Comment (tux3) [reply tankersdk_identity/test/test_identity.py]: Once'd, thank you!

**Review by blastrock — APPROVED** [review]

**Pull request merged**